### PR TITLE
AddChallenge: randomise validators

### PR DESCRIPTION
**Pull request by Sriep**
1. Randomise validators selected for a challenge; issue https://github.com/0chain/0chain/issues/234. 
2. Unit test for `addChallenge` to check these changes.
3. Removed dead comments.
4. Lint fixes:
```
smartcontract/storagesc/challenge.go:32:2: ineffectual assignment to idx (ineffassign)
	idx := -1
	^
smartcontract/storagesc/challenge_test.go:418:2: assign: self-assignment of blobber to blobber (govet)
	blobber = blobber
	^
smartcontract/storagesc/challenge_test.go:416:2: SA4006: this value of `blobber` is never used (staticcheck)
	blobber, err := ssc.getStakePool(blobberId, ctx)
	^
```
5. Reduce the lint limit.

### CODE REVIEW

File: code/go/0chain.net/smartcontract/storagesc/challenge_test.go [hunk 1]
```diff
@@ -25,6 +27,138 @@ const (
...
+				numValidators: 10,
+				dataShards:    4,
+				randomSeed:    1,
+			},
+			want: want{
+				validators: []int{6, 3, 8, 4},
+			},
+		},
+		{
+			name: "OK dataShars > validaors",
```
Comment (bbist) [inline code/go/0chain.net/smartcontract/storagesc/challenge_test.go:124]: Typo (maybe): 
dataShars -> dataShards
validaors -> validators

**Review by bbist — COMMENTED** [review]
Looks good.

File: code/go/0chain.net/smartcontract/storagesc/sc.go [hunk 2]
```diff
@@ -27,11 +27,16 @@ type StorageSmartContract struct {
 }
 
 func NewStorageSmartContract() (sci.SmartContractInterface, *sci.SmartContract) {
+	var sscCopy = newStorageSmartContract()
+	return sscCopy, sscCopy.SmartContract
+}
+
+func newStorageSmartContract() *StorageSmartContract {
```
Comment (andrenerd) [inline code/go/0chain.net/smartcontract/storagesc/sc.go:34]: sorry, what's that?
what was the reason to add a function with such "no-no" name, that is not used?
if it's needed, better to parametrize the original one.

ps. would be nice to rename "sscCopy" var to "scc". to reduce gradually "poor naming" cases.

Comment (Sriep) [reply code/go/0chain.net/smartcontract/storagesc/sc.go]: Removed dependency on sc.go

**Pull request merged**